### DISCUSSION TITLE
Microsoft.Data.Sqlite in WebAssembly not working

### DISCUSSION THREAD

**Discussion by AbdalaMask** [unoplatform/uno#7452] [Q&A]
### Current behavior

Microsoft.Data.Sqlite in WebAssembly not working
reading and writing

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

**Comment by AbdalaMask** [comment]
Microsoft.Data.Sqlite in WebAssembly not working

**Comment by AbdalaMask** [comment]
[SQLiteDatabase.zip](https://github.com/unoplatform/uno/files/7506825/SQLiteDatabase.zip)
![image](https://user-images.githubusercontent.com/37013638/140974999-8f1f86df-d67c-405a-aba9-0780b30011d9.png)

**Comment by AbdalaMask** [comment]
Microsoft.Data.Sqlite in WebAssembly not working

**Comment by AbdalaMask** [comment]
using Microsoft.Data.Sqlite" Version="5.0.11"
please help

**Reply by jeromelaban** [reply]
For SQLite use, you can follow the example provided here: https://github.com/unoplatform/Uno.Samples/tree/master/UI/SQLiteSample.

Microsoft.Data.Sqlite 5.0 is not supported by .NET, you'll need to use 6.0 or later. See https://github.com/unoplatform/Uno.SQLitePCLRaw.Wasm/tree/master/src/samples/EFCoreSample for an example.

**Reply by AbdalaMask** [reply]
i tried all
Why are you working?

**Reply by AbdalaMask** [reply]
I tried all and it didn't work

**Reply by AbdalaMask** [reply]
SQLiteSample WebAssembly not working 

I want an example of reading and writing data display

**Comment by AbdalaMask** [comment]
Microsoft.Data.Sqlite 5.0 working UWP and Gtk and wpf  and android

WebAssembly not working

**Reply by Daoting** [reply]
This may be because the file name referenced at initialization is incorrect. Wasm reference sqlite3. Other is e_sqlite3. Try it:
```
#if WASM
            // .net5.0 只能引用 SQLite3Provider_sqlite3，DllImport("sqlite3")
            // 默认为 SQLite3Provider_e_sqlite3 引用时出错！
            SQLitePCL.raw.SetProvider(new SQLitePCL.SQLite3Provider_sqlite3());
#else
            // 初始化不同平台的包绑定！V2支持类型和属性的绑定
            // 内部调用 SQLitePCL.raw.SetProvider(new SQLitePCL.SQLite3Provider_e_sqlite3());
            SQLitePCL.Batteries_V2.Init();
#endif
```

**Comment by AbdalaMask** [comment]
not working 
![image](https://user-images.githubusercontent.com/37013638/141497479-e31b1b2f-325f-4ece-8a07-6499c7d4c681.png)
SQLiteSample.Wasm 

help

**Comment by AbdalaMask** [comment]
try agen
![image](https://user-images.githubusercontent.com/37013638/141499660-45ad82c1-96e0-41dc-a414-447cb20566b8.png)
erorr

**Comment by AbdalaMask** [comment]
Help

**Comment by Daoting** [comment]
Add a reference `SQLitePCLRaw.provider.sqlite3`, only wasm project.

```
	<ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' ">
		<PackageReference Include="SQLitePCLRaw.provider.sqlite3" Version="2.0.4" />
	</ItemGroup>
```

**Comment by AbdalaMask** [comment]
net6 wasm project

**Comment by AbdalaMask** [comment]
![image](https://user-images.githubusercontent.com/37013638/141730627-708baa91-c793-4a53-8deb-0b94bbaf5684.png)

**Reply by AbdalaMask** [reply]
System.DllNotFoundException: sqlite3
![image](https://user-images.githubusercontent.com/37013638/141737153-d0afe131-32c7-4d2e-bbdb-d9465aea165f.png)

**Comment by AbdalaMask** [comment]
![image](https://user-images.githubusercontent.com/37013638/141737396-80bd11a0-3e6e-412e-942b-389f15b02df1.png)

**Comment by AbdalaMask** [comment]
uno-bootstrap.js:204 Error: System.DllNotFoundException: sqlite3
   at :5000/SQLitePCL.SQLite3Provider_sqlite3.SQLitePCL.ISQLite3Provider.sqlite3_libversion_number() in SQLitePCLRaw.provider.sqlite3.dll:token 0x600003e+0x0
   at SQLitePCL.raw.SetProvider(:5000/ISQLite3Provider imp) in SQLitePCLRaw.core.dll:token 0x6000021+0x8
   at SQLiteSample.Wasm.Program.Main(:5000/String[] args) in D:\MyPrograms\PLC C#\Uno.Samples\UI\SQLiteSample\SQLiteSample\SQLiteSample\SQLiteSample.Wasm\Program.cs:line 12
    at Object.call_method (uno-bootstrap.js:157)
    at Object.mainInit (uno-bootstrap.js:198)
    at checkDone (uno-bootstrap.js:451)
    at processDependency (uno-bootstrap.js:475)
    at Object.execCb (require.js:5)
    at e.check (require.js:5)
    at e.<anonymous> (require.js:5)
    at require.js:5
    at require.js:5
    at each (require.js:5)
mainInit @ uno-bootstrap.js:204

**Comment by AbdalaMask** [comment]
I tried everything
And the same problem
please solve

**Comment by Daoting** [comment]
It's as if you're missing steps

1. Make sure that the emsdk installation is successful. The current version is 2.0.23. The default installation path is as follows:
![image](https://user-images.githubusercontent.com/29876815/141873245-f16c3d8f-f680-46d7-9f5e-2f8d81c3713d.png)

2. Add a nuget package:
` <PackageReference Include="Uno.sqlite-wasm" Version="3.34.1.3" />`

Here's a Chinese document that I hope will help you.
[WebAssembly.docx](https://github.com/unoplatform/uno/files/7542428/WebAssembly.docx)

**Reply by AbdalaMask** [reply]
thank you for replying

**Comment by AbdalaMask** [comment]
![image](https://user-images.githubusercontent.com/37013638/143493120-94c1aa1a-2c90-4feb-9f88-7a0ecb3ff5d9.png)

how fix 
[HslStudio.zip](https://github.com/unoplatform/uno/files/7604958/HslStudio.zip)

**Comment by AbdalaMask** [comment]
show Video
[_.zip](https://github.com/unoplatform/uno/files/7604968/_.zip)

**Comment by AbdalaMask** [comment]
program exited (with status: 1), but EXIT_RUNTIME is not set, so halting execution but not exiting the runtime or preventing further async execution (build with EXIT_RUNTIME=1, if you want a true shutdown)

**Comment by AbdalaMask** [comment]
help
![image](https://user-images.githubusercontent.com/37013638/143627772-d7e248a4-b067-43e3-900e-716d624daa23.png)
